File: XamarinStore.iOS/Views/BottomButtonView.cs
using System;
using MonoTouch.UIKit;
using System.Drawing;

namespace XamarinStore
{
	public class BottomButtonView : BrightlyBlurredUIView
	{
		public readonly ImageButton Button;
		public Action ButtonTapped { get; set; }
		public string ButtonText
		{
			get{ return Button.Text; }
			set{ Button.Text = value; }
		}
		public const float Height = 75;

		public BottomButtonView()
		{
			this.AddSubview(Button = new  ImageButton());
			Button.Layer.BackgroundColor = Color.Green;
			Button.Layer.CornerRadius = 5f;
			Button.Font = UIFont.BoldSystemFontOfSize (UIFont.ButtonFontSize);
			Button.SizeToFit();
			Button.TouchUpInside += (object sender, EventArgs e) => ButtonTapped();
			this.TintColor = UIColor.White;
			this.AccentColorIntensity = 0f;
		}
		const float padding = 15f;
		public override void LayoutSubviews ()
		{
			base.LayoutSubviews ();
			var bound = Bounds;
			bound.X = padding;
			bound.Y = padding;
			bound.Width -= padding * 2;
			bound.Height -= padding * 2;
			Button.Frame = bound;
		}
	}
}

File: F#/XamarinStore.iOS/Views/BottomButtonView.fs
namespace XamarinStore

open System
open MonoTouch.UIKit
open MonoTouch.CoreGraphics
open System.Drawing

[<AllowNullLiteralAttribute>]
type BottomButtonView() as this =
    inherit BrightlyBlurredUIView()

    let padding = 15.0f
    let button = new ImageButton()

    do this.AddSubview button
       button.Layer.BackgroundColor <- Color.Green.ToUIColor().CGColor
       button.Layer.CornerRadius <- 5.0f
       button.Font <- UIFont.BoldSystemFontOfSize (UIFont.ButtonFontSize)
       button.SizeToFit()
       button.TouchUpInside.Add( fun _ -> this.ButtonTapped() )
       this.TintColor <- UIColor.White
       this.AccentColorIntensity <- 0.0f
    
    member val Button = button with get
    member val ButtonTapped = fun ()->() with get,set
    member this.ButtonText
        with get () = button.Text
        and set value = button.Text <- value

    static member val Height = 75.0f with get

    override this.LayoutSubviews () =
        base.LayoutSubviews ()
        let mutable bound = this.Bounds
        bound.X <- padding
        bound.Y <- padding
        bound.Width <- bound.Width - padding * 2.0f
        bound.Height <- bound.Height - padding * 2.0f
        button.Frame <- bound
